src/component/mxgraph/shape/event-shapes.ts

```diff
@@ -81,12 +81,34 @@ abstract class EventShape extends mxEllipse {
     const paintIcon = this.iconPainters.get(StyleUtils.getBpmnEventKind(this.style)) || (() => this.iconPainter.paintEmptyIcon());
     paintIcon(paintParameter);
   }
+
+  protected setDashedOuterShapePattern(paintParameter: PaintParameter, isInterrupting: string): void {
+    if (isInterrupting === 'false') {
+      paintParameter.c.setDashed(true, false);
+      paintParameter.c.setDashPattern('3 2');
+    }
+  }
+
+  protected restoreOriginalOuterShapePattern(paintParameter: PaintParameter): void {
+    // Restore original configuration
+    // TODO missing mxShape.configureCanvas in mxgraph-type-definitions (this will replace explicit function calls)
+    // this.configureCanvas(c, x, y, w, h);
+    paintParameter.c.setDashed(StyleUtils.isDashed(this.style), StyleUtils.isFixDash(this.style));
+    paintParameter.c.setDashPattern(StyleUtils.getDashPattern(this.style));
+  }
 }
 
 export class StartEventShape extends EventShape {
   public constructor(bounds: mxRectangle, fill: string, stroke: string, strokewidth: number = StyleDefault.STROKE_WIDTH_THIN) {
     super(bounds, fill, stroke, strokewidth);
   }
+
+  protected paintOuterShape(paintParameter: PaintParameter): void {
```

Comment:
:information_source: I like the new code sharing introduced here.
However, we sill have code duplicated between `StartEventShape` and `BoundaryEventShape` as their `` method implementation is exactly the same.
```typescript
  protected paintOuterShape(paintParameter: PaintParameter): void {
    const isInterrupting = StyleUtils.getBpmnIsInterrupting(this.style);
    super.setDashedOuterShapePattern(paintParameter, isInterrupting);
    super.paintOuterShape(paintParameter);
    super.restoreOriginalOuterShapePattern(paintParameter);
  }
```
:question: Why not moving everything to the base `EventShape`. I tested the following quickly, this seems working
```typescript
  public paintVertexShape(c: mxAbstractCanvas2D, x: number, y: number, w: number, h: number): void {
    this.markNonFullyRenderedEvents(c);
    const paintParameter = buildPaintParameter(c, x, y, w, h, this, 0.25, this.withFilledIcon);

    this.setDashedOuterShapePattern(paintParameter, StyleUtils.getBpmnIsInterrupting(this.style));
    this.paintOuterShape(paintParameter);
    this.restoreOriginalOuterShapePattern(paintParameter);

    this.paintInnerShape(paintParameter);
  }
```